src/main/kotlin/org/rust/ide/inspections/lints/RsUnusedImportInspection.kt

```diff
@@ -18,32 +19,79 @@ class RsUnusedImportInspection : RsLintInspection() {
     override fun getLint(element: PsiElement): RsLint = RsLint.UnusedImports
 
     override fun buildVisitor(holder: RsProblemsHolder, isOnTheFly: Boolean) = object : RsVisitor() {
-        override fun visitUseSpeck(o: RsUseSpeck) {
+        override fun visitUseItem(item: RsUseItem) {
             if (!holder.project.isNewResolveEnabled) return
-            val item = o.ancestorStrict<RsUseItem>() ?: return
             if (item.isReexport) return
-            if (o.path?.resolveStatus != PathResolveStatus.RESOLVED) return
 
             // Do not check uses if there is a child mod inside the current mod
-            val parentMod = o.containingMod
+            val parentMod = item.containingMod
             val hasChildMods = parentMod.expandedItemsExceptImplsAndUses.any {
                 it is RsModItem || it is RsModDeclItem
             }
-            if (o.parentOfType<RsFunction>() == null && hasChildMods) return
+            if (item.parentOfType<RsFunction>() == null && hasChildMods) return
 
-            checkUseSpeck(o, holder)
+            val owner = item.parent as? RsItemsOwner ?: return
+            val usage = owner.pathUsage
+
+            val speck = item.useSpeck ?: return
+
+            val isSpeckUsedMap = HashMap<RsUseSpeck, Boolean>()
+            storeUseSpeckUsage(speck, usage, isSpeckUsedMap)
+            markUnusedSpecks(speck, isSpeckUsedMap, holder)
```

Comment:
It's a bit inconsistent that `storeUseSpeckUsage` is outside class and `markUnusedSpecks` is inside (since these methods are highly related)

Reply:
Right, `markUnusedSpecks` needed to be inside the class because of `registerLintProblem`. I'll move `storeUseSpeckUsage` inside the class too.